python-scripts/eeprom-util.py

```diff
@@ -0,0 +1,198 @@
+    self.gateway = bytes(eepFile.read(4))         # default gateway
+    self.subnet = bytes(eepFile.read(4))          # subnet
+    self.unknown4 = bytes(eepFile.read(4))        # TBD
+    self.dvdReg = bytes(eepFile.read(4))          # DVD region code
+    self.unknown5 = bytes(eepFile.read(64))       # TBD
+
+# Because of the confounder we have to split the RC4 decryption / encryption proccess
+# into two functions (eepKeyPrepCrypt / eepKeyCrypt). We also need a way to share state
+# between the the two functions (EepKeyObj). In a nutshell we fill and EepKeyObj method
+# object with the initial state by calling eepKeyPrepCrypt. Then we call eepKeyCrypt
+# on the confounder data. eepKeyCrypt will push state data back into the EepKeyObj method 
+# object (i / j) which is necessary when we call eepKeyCrypt on the next chunk of data
+# from the Eeprom method object (eeprom.hddKey + eeprom.gameReg). This is also why we
+# don't use any standard RC4 crypto libaries here.
+
+class EepKeyObj:
+  state = bytearray(range(256))
+  i = 0
+  j = 0
+
+def eepKeyPrepCrypt(eepKey, eepKeyObj):
+  j = 0
+
+  for i in range(256):
+    j = (j + eepKeyObj.state[i] + eepKey[i % len(eepKey)]) % 256
+    eepKeyObj.state[i], eepKeyObj.state[j] = eepKeyObj.state[j], eepKeyObj.state[i]
+
+def eepKeyCrypt(eepCryptData, eepKeyObj):
+  eepData = bytearray()
+  i = eepKeyObj.i
+  j = eepKeyObj.j
+  xorInd = 0
+
+  for byte in eepCryptData:
+    i = (i + 1) % 256
+    j = (j + eepKeyObj.state[i]) % 256
+    eepKeyObj.state[i], eepKeyObj.state[j] = eepKeyObj.state[j], eepKeyObj.state[i]
+    eepData.append(byte ^ eepKeyObj.state[(eepKeyObj.state[i] + eepKeyObj.state[j]) % 256])
+
+  eepKeyObj.i = i
+  eepKeyObj.j = j
+  return eepData
+
+parser = optparse.OptionParser()
+parser.add_option('-f', action='store', dest='eepFileLoc')
+parser.add_option('-k', action='store', dest='keyFileLoc')
+(options, args) = parser.parse_args()
+
+if options.eepFileLoc:
+  eeprom = Eeprom(options.eepFileLoc)
+  print ('\033[1mEncrypted EEPROM Data\033[0m')
+  print ('EEPROM Hash:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.hash)))
+  print ('Encrypted Confounder:\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.confounder)))
+  print ('Encrypted HDD Key:\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.hddKey)))
+  print ('Encrypted Game Region Code:\t{}'.format(''.join(format(x, '02x') for x in eeprom.gameReg)))
+
+  print ('\n\033[1mIdentity EEPROM Data\033[0m')
+  print ('Identity Checksum:\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.identChk)))
+  print ('Serial Number:\t\t\t{}'.format(eeprom.serial.decode('ascii')))
+  print ('MAC Address:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.mac)))
+  print ('Online Key:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.onlineKey)))
+
+  # print NTSC / PAL instead of their respective codes
+  if ''.join(format(x, '02x') for x in eeprom.videoStd) == '00014000':
```

Comment:
changed to int compare, python byte notation is ugly look. With a few comments it's easy to know what's going on. I probably did them as strings originally because you could easily see what it was doing.